### ISSUE TITLE
VObject is not thread-safe

### ISSUE THREAD

**Issue by Unrud** [Kozea/Radicale#675]
The problem seems to be in [dateutil](https://github.com/dateutil/dateutil/).
```
Traceback (most recent call last):
  File "radicale/storage.py", line 1357, in _get_with_metadata
    ctext = vobject_item.serialize()
  File "vobject/base.py", line 254, in serialize
    return behavior.serialize(self, buf, lineLength, validate)
  File "vobject/behavior.py", line 155, in serialize
    cls.generateImplicitParameters(obj)
  File "vobject/icalendar.py", line 971, in generateImplicitParameters
    findTzids(obj, tzidsUsed)
  File "vobject/icalendar.py", line 969, in findTzids
    findTzids(child, table)
  File "vobject/icalendar.py", line 969, in findTzids
    findTzids(child, table)
  File "object/icalendar.py", line 964, in findTzids
    tzid = TimezoneComponent.registerTzinfo(tzinfo)
  File "vobject/icalendar.py", line 129, in registerTzinfo
    tzid = obj.pickTzid(tzinfo)
  File "vobject/icalendar.py", line 334, in pickTzid
    if tzinfo is None or (not allowUTC and tzinfo_eq(tzinfo, utc)):
  File "vobject/icalendar.py", line 1996, in tzinfo_eq
    if not dt_test(datetime.datetime(startYear, 1, 1)):
  File "vobject/icalendar.py", line 1994, in dt_test
    return tzinfo1.utcoffset(dt) == tzinfo2.utcoffset(dt)
  File "dateutil/tz/tz.py", line 1109, in utcoffset
    return self._find_comp(dt).tzoffsetto
  File "dateutil/tz/tz.py", line 1062, in _find_comp
    return self._cachecomp[self._cachedate.index((dt, self._fold(dt)))]
IndexError: list index out of range
```

Edit: https://github.com/dateutil/dateutil/issues/428

**Comment by prlw1** [comment]
While on VObject, I observe:

```
Python 3.6.1 (default, Jun 27 2017, 15:00:07) 
[GCC 5.4.0] on netbsd8
Type "help", "copyright", "credits" or "license" for more information.
>>> import vobject
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'vobject'
>>> import vdirsyncer.vobject
>>> exit()

```
So I need the trivial s/import vobject/import vdirsyncer.vobject/ in \_\_init__.py and radicale_storage.py

**Comment by Unrud** [comment]
> So I need the trivial s/import vobject/import vdirsyncer.vobject/ in __init__.py and radicale_storage.py

No, ``vobject`` and ``vdirsyncer.vobject`` are not the same. You have to install the ``vobject`` python module. If you install Radicale with ``$ python3 -m pip install --upgrade radicale`` it should install all dependencies.

**Comment by Unrud** [comment]
Fixed in https://github.com/dateutil/dateutil/commit/94f1639b513280ca73144bc494d826616be22768. This can be closed, when the next version of ``dateutil`` gets released.

**Comment by Unrud** [comment]
Fixed by #813
